### PR TITLE
ROC-3140: Consistent use of usergroup and other changes

### PR DESCRIPTION
**Pull request by singh-surjeet**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-3140

### Change description ###
Renaming role variable to usergroup, amending README & pointing to latest IDAM_API docker image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: bin/create-citizen-user.sh [hunk 1]
```diff
@@ -6,8 +6,8 @@ USER_EMAIL="${1:-me@server.net}"
 FORENAME="${2:-John}"
 SURNAME="${3:-Smith}"
 PASSWORD=Password12
-ROLE="cmc-private-beta"
+USER_GROUP="cmc-private-beta"
```
Comment (geek-soft) [inline bin/create-citizen-user.sh]: I do not think that change is worth having. I would suggest reverting it.

**Comment by geek-soft** [comment]
@singh-surjeet when merging please apply https://chris.beams.io/posts/git-commit/ guidelines. Best to start by changing PR name.

File: README.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ https://github.com/SeleniumHQ/docker-selenium#debugging
 The first time you will need to create a user to login with
 Run: 
 ```bash
-$ ./bin/create-idam-user.sh [optional-email] [optional-firstname] [optional-lastname]
+$ ./bin/create-user.sh [optional-email] [optional-firstname] [optional-lastname]
```
Comment (geek-soft) [inline README.md]: Good spot. Worth mentioning fourth additional argument though.

File: bin/create-citizen-user.sh (hunk 1 above)
Comment (geek-soft) [reply bin/create-citizen-user.sh]: If you want to keep it please make that change in other places as well as `integration-tests/docker/ccd-definition-import/create-importer-user.sh`

File: bin/create-user.sh [hunk 3]
```diff
@@ -6,19 +6,19 @@ USER_EMAIL="${1:-me@server.net}"
...
-      "code": "'${ROLE}'"
+      "code": "'${USER_GROUP}'"
     },
     "activationDate": "",
     "lastAccess": "",
     "password": "'${PASSWORD}'"
 }'
 
-echo -e "Created user with:\nUsername: ${USER_EMAIL}\nPassword:${PASSWORD}\nFirstname: ${FORENAME}\nSurname: ${SURNAME}\nROLE: ${ROLE}"
+echo -e "Created user with:\nUsername: ${USER_EMAIL}\nPassword:${PASSWORD}\nFirstname: ${FORENAME}\nSurname: ${SURNAME}\nUserGroup: ${USER_GROUP}"
```
Comment (geek-soft) [inline bin/create-user.sh]: If you want tot progress with rename please have `User group:` here.

File: docker-compose.yml [hunk 4]
```diff
@@ -125,7 +125,7 @@ services:
         idam-api:
           condition: service_started
     idam-api:
-      image: docker.artifactory.reform.hmcts.net/auth/idam-api:ea84d227d19ae816686407bb49e747f779def5e4
+      image: docker.artifactory.reform.hmcts.net/auth/idam-api:a0b158c188c9179910f9fbde89ea9e2ca30694b9
```
Comment (geek-soft) [inline docker-compose.yml]: Is that version stable? I know that IDAM is under active dev right now. I do not see a reason to change the container right now. We can swap to use `citizen` role without changing IDAM version.

Comment (singh-surjeet) [reply docker-compose.yml]: For this change probably we don't need to update version. Reverting it...

File: README.md (hunk 2 above)
Comment (singh-surjeet) [reply README.md]: Fine.

File: bin/create-citizen-user.sh (hunk 1 above)
Comment (singh-surjeet) [reply bin/create-citizen-user.sh]: I think I rather keep this change and apply at other places as well.

Comment (geek-soft) [reply bin/create-citizen-user.sh]: Do you know if FrIDAM has the same concept of user group? Roles is more flexible IMO.

File: docker-compose.yml (hunk 4 above)
Comment (geek-soft) [reply docker-compose.yml]: When reverted please update PR name and do not forget to remove it from squash commit message at merge time.

File: bin/create-citizen-user.sh (hunk 1 above)
Comment (singh-surjeet) [reply bin/create-citizen-user.sh]: FrIDAM is planning to remove user group I assume. Don't know if it's a good idea if we want to introduce role based AC in near future for selected access.

Comment (singh-surjeet) [reply bin/create-citizen-user.sh]: actioned

File: bin/create-user.sh (hunk 3 above)
Comment (singh-surjeet) [reply bin/create-user.sh]: actioned

File: docker-compose.yml (hunk 4 above)
Comment (singh-surjeet) [reply docker-compose.yml]: Reverted

File: bin/create-citizen-user.sh (hunk 1 above)
Comment (geek-soft) [reply bin/create-citizen-user.sh]: @singh-surjeet if FrIDAM is going to remove concept of user groups why you want to change the naming to something that will become outdated soon? Role word fits in the current model just alright.

Comment (singh-surjeet) [reply bin/create-citizen-user.sh]: They might not if they think of the lines of RBAC. IMO roles must be grouped other they're all over the place. These changes reflect current state and we'll worry about Fridam they're ready.

**Pull request merged**